panic: slice bounds out of range during provider restart

**Log**
```
(info)  [provider] Import private key from: /private/private.key
(info)  [provider] Staring provider with address: e50a943bea876f1747aafa84f0aef53feeb668b47b607be6d68030ea1c96d31f
D[2019-10-18|11:43:57.212] using kube client                            app=akash
I[2019-10-18|11:43:57.212] using in-cluster config                      app=akash cmp=cluster-client
I[2019-10-18|11:43:57.391] found leases                                 module=provider-cluster app=akash cmp=service num-active=5 num-skipped=41
D[2019-10-18|11:43:57.391] deployment                                   module=provider-cluster app=akash cmp=service lease=18daf58231c84cd4738b9a635234b7eac2716606fbb3f6decf1a53cf0736cb58/1/2/e50a943bea876f1747aafa84f0aef53feeb668b47b607be6d68030ea1c96d31f mgroup=global
D[2019-10-18|11:43:57.391] deployment                                   module=provider-cluster app=akash cmp=service lease=9e849074d6f9190e3809f9e68ee4953532eab1bfffd5e207308535f3e911a2c1/1/2/e50a943bea876f1747aafa84f0aef53feeb668b47b607be6d68030ea1c96d31f mgroup=global
D[2019-10-18|11:43:57.391] deployment                                   module=provider-cluster app=akash cmp=service lease=3d06c8bb6f849d2dc4824231daef0fad456f62fc22a8682c7213a62bd4df0632/1/2/e50a943bea876f1747aafa84f0aef53feeb668b47b607be6d68030ea1c96d31f mgroup=global
panic: runtime error: slice bounds out of range

goroutine 1 [running]:
github.com/ovrclk/akash/provider/cluster.findDeployments(0x1be1280, 0xc000f9ff00, 0x1be1f80, 0xc00182d500, 0x7f197030e4b8, 0xc000fafb60, 0x1be78c0, 0xc000f9ff80, 0x2, 0x38, ...)
	/home/travis/gopath/src/github.com/ovrclk/akash/provider/cluster/service.go:292 +0xdf9
github.com/ovrclk/akash/provider/cluster.NewService(0x1be1280, 0xc000f9ff00, 0x1be78c0, 0xc000f9ff80, 0x1be14c0, 0xc000fafda0, 0x7f197030e4b8, 0xc000fafb60, 0x203000, 0x203000, ...)
	/home/travis/gopath/src/github.com/ovrclk/akash/provider/cluster/service.go:52 +0x42a
github.com/ovrclk/akash/provider.NewService(0x1be1280, 0xc000f9ff00, 0x1be78c0, 0xc000f9ff80, 0x1be14c0, 0xc000fafda0, 0x7f197030e4b8, 0xc000fafb60, 0x1be1280, 0xc000f9fcc0, ...)
	/home/travis/gopath/src/github.com/ovrclk/akash/provider/service.go:42 +0x15c
main.doProviderRunCommand(0x1c17460, 0xc000b1a000, 0xc000b0f400, 0xc000afd020, 0x1, 0x6, 0x0, 0x0)
	/home/travis/gopath/src/github.com/ovrclk/akash/cmd/akash/provider.go:207 +0x98e
github.com/ovrclk/akash/cmd/akash/session.RequireHost.func1(0x1c17460, 0xc000b1a000, 0xc000b0f400, 0xc000afd020, 0x1, 0x6, 0x19f9b40, 0xc00046a100)
	/home/travis/gopath/src/github.com/ovrclk/akash/cmd/akash/session/session.go:96 +0xee
github.com/ovrclk/akash/cmd/akash/session.RequireNode.func1(0x1c17460, 0xc000b1a000, 0xc000b0f400, 0xc000afd020, 0x1, 0x6, 0x1bf1e60, 0xc00046a100)
	/home/travis/gopath/src/github.com/ovrclk/akash/cmd/akash/session/session.go:123 +0x134
github.com/ovrclk/akash/cmd/akash/session.WithSession.func1.1(0x1be1280, 0xc000955dc0, 0x0, 0x0)
	/home/travis/gopath/src/github.com/ovrclk/akash/cmd/akash/session/session.go:67 +0x1b4
github.com/ovrclk/akash/cmd/common.RunForeverWithContext(0x1be1280, 0xc000955dc0, 0xc000b12a20, 0xc000319d48, 0x0)
	/home/travis/gopath/src/github.com/ovrclk/akash/cmd/common/signal.go:35 +0x91
github.com/ovrclk/akash/cmd/common.RunForever(...)
	/home/travis/gopath/src/github.com/ovrclk/akash/cmd/common/signal.go:27
github.com/ovrclk/akash/cmd/akash/session.WithSession.func1(0xc000b0f400, 0xc000afd020, 0x1, 0x6, 0x0, 0x0)
	/home/travis/gopath/src/github.com/ovrclk/akash/cmd/akash/session/session.go:56 +0xaf
github.com/ovrclk/akash/vendor/github.com/spf13/cobra.(*Command).execute(0xc000b0f400, 0xc000afcf60, 0x6, 0x6, 0xc000b0f400, 0xc000afcf60)
	/home/travis/gopath/src/github.com/ovrclk/akash/vendor/github.com/spf13/cobra/command.go:756 +0x465
github.com/ovrclk/akash/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc000af0500, 0xc000319f80, 0x1, 0x1)
	/home/travis/gopath/src/github.com/ovrclk/akash/vendor/github.com/spf13/cobra/command.go:846 +0x2c0
github.com/ovrclk/akash/vendor/github.com/spf13/cobra.(*Command).Execute(...)
	/home/travis/gopath/src/github.com/ovrclk/akash/vendor/github.com/spf13/cobra/command.go:794
main.main()
	/home/travis/gopath/src/github.com/ovrclk/akash/cmd/akash/main.go:26 +0x372
```

--- a/provider/cluster/service.go
+++ b/provider/cluster/service.go
@@ -283,18 +283,17 @@ func findDeployments(ctx context.Context, log log.Logger, client Client, session
 
 	log.Info("found leases", "num-active", len(leases), "num-skipped", len(leaseList.Items)-len(leases))
 
-	dcount := len(deployments)
-	for idx, deployment := range deployments {
+	active := make([]Deployment, 0, len(deployments))
+
+	for _, deployment := range deployments {
 		if _, ok := leases[deployment.LeaseID().Path()]; !ok {
 			continue
 		}
-
-		deployments = append(deployments[:idx], deployments[idx+1:]...)
-
+		active = append(active, deployment)
 		log.Debug("deployment", "lease", deployment.LeaseID(), "mgroup", deployment.ManifestGroup().Name)
 	}
 
-	log.Info("found deployments", "num-active", len(deployments), "num-skipped", dcount-len(deployments))
+	log.Info("found deployments", "num-active", len(active), "num-skipped", len(deployments)-len(active))
 
-	return deployments, nil
+	return active, nil
 }

